### ISSUE TITLE
Issue with calculation of `union_rect` in `FloatOverlay::build_graph_with_solver` when `clip_paths` is empty

### ISSUE THREAD

**Issue by MatchaChoco010** [iShape-Rust/iOverlay#4]
I found a issue when using `ShapeType::Subject` in `FloatOverlay` and would like to report it.

I added only `ShapeType::Subject` paths for `FloatOverlay` and tried to do `graph.extract_shapes` with `OverlayRule::Subject`, but the calculation does not seem to be correct.

```rust
overlay.add_paths(&shape, ShapeType::Subject);  
let graph = overlay.build_graph(FillRule::NonZero);
let shapes = graph.extract_shapes(OverlayRule::Subject);
```

In the library code, `src/core/float_overlay.rs` calculates `union_rect` from `subj_rect` and `clip_rect`, but there seems to be a problem in this part.
When `subj_rect` and `clip_rect` are created with `F64Rect::with_shape`, if empty paths are inserted, rects with `-f64::MAX` min and max values are created.
Calculating `union_rect` based on these all `-f64::MAX` rects will result in an incorrect rect, and since `PointAdapter` is calculated based on it, the process will be incorrect.

To fix the above problem, it seems that if `self.subj_paths` or `self.clip_paths` is empty, it is necessary to create a `PointAdapter` by calculating the rect with only the other one.

More to the point, it might be better to return `Result` `Err` if `build_graph` is called with both `self.subj_paths` and `self.clip_paths` empty.

**Comment by NailxSharipov** [comment]
Good job to find this issue.

Yes correct library will not supposed to work with paths less 3 points.
We should do something with it. I saw your issue with fixing `F64Rect` constructor let me explaining my thoughts there.
For this issue a create new empty correctness tests at [float_overlay_tests.rs](https://github.com/iShape-Rust/iOverlay/blob/main/tests/float_overlay_tests.rs)

**Comment by NailxSharipov** [comment]
fixed in 1.2.3

**Comment by NailxSharipov** [comment]
+
